polls/views.py

```diff
@@ -0,0 +1,40 @@
+from django.shortcuts import get_object_or_404, render
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+from django.db.models import F
+from django.views import generic
+
+from .models import Question, Choice
+
+
+class IndexView(generic.ListView):
+    template_name = 'polls/index.html'
+    context_object_name = 'latest_question_list'
+
+    def get_queryset(self):
+        return Question.objects.order_by('-pub_date')[:5]
+
+
+class DetailView(generic.DetailView):
+    model = Question
+    template_name = 'polls/detail.html'
+
+
+class ResultsView(generic.DetailView):
+    model = Question
+    template_name = 'polls/results.html'
+
+
+def vote(request, question_id):
+    question = get_object_or_404(Question, pk=question_id)
+    try:
+        selected_choice = question.choice_set.get(pk=request.POST['choice'])
+    except (KeyError, Choice.DoesNotExist):
+        return render(request, 'polls/detail.html', {
+            'question': question,
+            'error_message': "You didn't select a choice"
+        })
+    else:
+        selected_choice.votes = F('votes') + 1
```

Comment:
Question:
why do you use query expression here instead of pure accumulation?
`selected_choice.votes +=1`

Reply:
In case two users vote at the same time. The F() method takes care of that issue according to documentation to have the database update that value instead of Python